vl53v1x/vl53l1x.go

```diff
@@ -0,0 +1,541 @@
+}
+
+// getRangingData stores in the buffer the ranging data
+func (d *Device) getRangingData() {
+	d.rangingData.mm = uint16((uint32(d.results.mmCrosstalkSD0)*2011 + 0x0400) / 0x0800)
+	switch d.results.status {
+	case 17: // MULTCLIPFAIL
+	case 2: // VCSELWATCHDOGTESTFAILURE
+	case 1: // VCSELCONTINUITYTESTFAILURE
+	case 3: // NOVHVVALUEFOUND
+		d.rangingData.status = HardwareFail
+		break
+
+	case 13: // USERROICLIP
+		d.rangingData.status = MinRangeFail
+		break
+
+	case 18: // GPHSTREAMCOUNT0READY
+		d.rangingData.status = SynchronizationInt
+		break
+
+	case 5: // RANGEPHASECHECK
+		d.rangingData.status = OutOfBoundsFail
+		break
+
+	case 4: // MSRCNOTARGET
+		d.rangingData.status = SignalFail
+		break
+
+	case 6: // SIGMATHRESHOLDCHECK
+		d.rangingData.status = SignalFail
+		break
+
+	case 7: // PHASECONSISTENCY
+		d.rangingData.status = WrapTargetFail
+		break
+
+	case 12: // RANGEIGNORETHRESHOLD
+		d.rangingData.status = XtalkSignalFail
+		break
+
+	case 8: // MINCLIP
+		d.rangingData.status = RangeValidMinRangeClipped
+		break
+
+	case 9: // RANGECOMPLETE
+		if d.results.streamCount == 0 {
+			d.rangingData.status = RangeValidNoWrapCheckFail
+		} else {
+			d.rangingData.status = RangeValid
+		}
+		break
+
+	default:
+		d.rangingData.status = None
+	}
+
+	d.rangingData.signalRateMCPS = float32(d.results.signalRateCrosstalkMCPSSD0) / (1 << 7)
+	d.rangingData.ambientRateMCPS = float32(d.results.ambientRateMCPSSD0) / (1 << 7)
+}
+
+// setupManualCalibration configures the manual calibration
+func (d *Device) setupManualCalibration() {
+	// save original VHV configs
+	d.VHVInit = d.readReg(VHV_CONFIG_INIT)
+	d.VHVTimeout = d.readReg(VHV_CONFIG_TIMEOUT_MACROP_LOOP_BOUND)
+
+	// disable VHV init
+	d.writeReg(VHV_CONFIG_INIT, d.VHVInit&0x7F)
+
+	// set loop bound to tuning param
+	d.writeReg(VHV_CONFIG_TIMEOUT_MACROP_LOOP_BOUND, (d.VHVTimeout&0x03)+(3<<2))
+
+	// override phasecal
+	d.writeReg(PHASECAL_CONFIG_OVERRIDE, 0x01)
+	d.writeReg(CAL_CONFIG_VCSEL_START, d.readReg(PHASECAL_RESULT_VCSEL_START))
+}
+
+// StartContinuous starts the continuous sensing mode
+func (d *Device) StartContinuous(periodMs uint32) {
+	d.writeReg32Bit(SYSTEM_INTERMEASUREMENT_PERIOD, periodMs*uint32(d.oscillatorOffset))
+	d.writeReg(SYSTEM_INTERRUPT_CLEAR, 0x01) // sys_interrupt_clear_range
+	d.writeReg(SYSTEM_MODE_START, 0x40)      // mode_range_timed
+}
+
+// StopContinuous stops the continuous sensing mode
+func (d *Device) StopContinuous() {
+	d.writeReg(SYSTEM_MODE_START, 0x80) // mode_range_abort
+
+	d.calibrated = false
+
+	// restore vhv configs
+	if d.VHVInit != 0 {
+		d.writeReg(VHV_CONFIG_INIT, d.VHVInit)
+	}
+	if d.VHVTimeout != 0 {
+		d.writeReg(VHV_CONFIG_TIMEOUT_MACROP_LOOP_BOUND, d.VHVTimeout)
+	}
+
+	// remove phasecal override
+	d.writeReg(PHASECAL_CONFIG_OVERRIDE, 0x00)
+}
+
+// writeReg sends a single byte to the specified register address
+func (d *Device) writeReg(reg uint16, value uint8) {
```

Comment:
Just wondering if this function and the other similar ones would be better added to the main repo like https://github.com/tinygo-org/tinygo/blob/master/src/machine/i2c.go#L17 and https://github.com/tinygo-org/tinygo/blob/master/src/machine/i2c.go#L30 since they are not really driver specific.

What do you think?

Reply:
I think it makes sense to have Write16BitRegister/Read16BitRegister, it's the first device/sensor I see that uses 16 bit registers, I'm not sure how common it is.

A bit off-topic, I don't know if there should be functions to convert (u)int16/32 <=> []byte{} (big/little endian) too. *binary* package is not (yet?) supported.

Reply:
Actually `binary` is currently working.

Reply:
oh, good to know, I need to make a few updates

Reply:
I think that would be useful. For now, we can leave them here and in parallel add these new methods to the `I2C` object, updating this code with the next release if necessary.